File: frontend/lib/models/user_dto.dart
class UserDto {
  final String kakaoId;
  final String nickname;
  final String profileImageUrl;

  UserDto({
    required this.kakaoId,
    required this.nickname,
    required this.profileImageUrl,
  });

  Map<String, dynamic> toJson() => {
    'kakao_id': kakaoId,
    'nickname': nickname,
    'profile_image_url': profileImageUrl,
  };
}

File: backend/src/main/java/net/skhu/dto/UserDto.java
package net.skhu.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

@Data
public class UserDto {
    int id;

    @JsonProperty("kakao_id")
    String kakaoId;

    String nickname;

    @JsonProperty("profile_image_url")
    String profileImageUrl;
}
